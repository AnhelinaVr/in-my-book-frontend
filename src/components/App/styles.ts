import { SxProps, Theme } from '@mui/material';

export const getStyles = (theme: Theme): Record<string, SxProps> => ({
  container: {
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
  },
});
