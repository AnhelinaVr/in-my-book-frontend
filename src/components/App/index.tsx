import { Box, useTheme } from '@mui/material';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Footer } from '../Footer';
import { Header } from '../Header';
import Home from '../../pages/Home';
import { Paths } from '../../constants';
import { getStyles } from './styles';

const links = [
  { path: Paths.home, name: 'Home' },
  { path: Paths.home, name: 'Rooms' },
  { path: Paths.home, name: 'Books' },
];

const App = () => {
  const theme = useTheme();
  const styles = getStyles(theme);

  // TODO: alert wrapper
  return (
    <Box sx={styles.container}>
      <BrowserRouter>
        <Header isAuthenticated={false} links={links} />
        <Routes>
          <Route path={Paths.home} element={<Home />} />
        </Routes>
        <Footer />
      </BrowserRouter>
    </Box>
  );
};

export default App;
