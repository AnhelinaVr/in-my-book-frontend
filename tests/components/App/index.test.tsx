import App from '../../../src/components/App';
import React from 'react';
import { render } from '@testing-library/react';

describe('<App />', () => {
  it('should match snapshot', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
