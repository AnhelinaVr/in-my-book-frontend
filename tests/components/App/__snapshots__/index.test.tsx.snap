// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1ab1jlb"
  >
    <div
      class="MuiBox-root css-cfepnb"
    >
      <div
        class="MuiBox-root css-yvr2s2"
      >
        <svg
          data-file-name="SvgLogo"
        />
      </div>
      <div
        class="MuiBox-root css-xd6fag"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1t1zlgp-MuiTypography-root-MuiLink-root"
          href="/"
        >
          Home
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1t1zlgp-MuiTypography-root-MuiLink-root"
          href="/"
        >
          Rooms
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1t1zlgp-MuiTypography-root-MuiLink-root"
          href="/"
        >
          Books
        </a>
      </div>
      <div
        class="MuiBox-root css-1fpwgg0"
        data-testid="header-registration"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Sign In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ah646k-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Sign Up
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-1qz38mj"
    >
      <img
        alt="background"
        src="main.jpg"
        style="position: absolute; width: 100%; left: 0px; top: 0px; object-fit: cover; height: 100%;"
      />
      <h2
        class="MuiTypography-root MuiTypography-h2 css-ustah6-MuiTypography-root"
      >
        BOOKS
      </h2>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-10dqq5q-MuiTypography-root"
      >
        club
      </h2>
      <h3
        style="margin-top: 92px;"
      >
        This application allows you to unite people with a common interest - reading books. The functionality allows users to share their opinions by writing book reviews, allows you to create rooms for joint discussion of books with the possibility of creating a video chat. In addition, the functionality of the application allows you to make personal notes about the book you read
      </h3>
    </div>
    <div
      class="MuiBox-root css-1lh2ipp"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-18zrctw-MuiTypography-root"
      >
        About Us
      </h2>
      <div
        class="MuiBox-root css-t68gks"
      />
      <div
        class="MuiBox-root css-kx25ay"
      >
        <div
          class="MuiBox-root css-1r3t2am"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-3jtalt-MuiTypography-root"
          >
            Join
          </h3>
          <img
            alt="card"
            src="function SvgImage1(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage1.name})
  );
}"
            style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Lorem Ipsum
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-1r3t2am"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-3jtalt-MuiTypography-root"
          >
            Communicate
          </h3>
          <img
            alt="card"
            src="function SvgImage2(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage2.name})
  );
}"
            style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Lorem Ipsum
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-1r3t2am"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-3jtalt-MuiTypography-root"
          >
            Have fun!
          </h3>
          <img
            alt="card"
            src="function SvgImage3(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgImage3.name})
  );
}"
            style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover;"
          />
          <p
            class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1ckkjjs-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Lorem Ipsum
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1b3cf6i"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-18zrctw-MuiTypography-root"
      >
        Our Heroes
      </h2>
      <div
        class="MuiBox-root css-1kmst6o"
      >
        <div
          class="MuiBox-root css-kx25ay"
        >
          <div
            class="MuiBox-root css-1r3t2am"
          >
            <img
              alt="card"
              src="anhelina_vrubleuskaya.jpg"
              style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 16px;"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
            >
              Anhelina Vrubleuskaya
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
            </p>
          </div>
          <div
            class="MuiBox-root css-1r3t2am"
          >
            <img
              alt="card"
              src="olga_polyanskaya.jpg"
              style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 16px;"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
            >
              Olga Polyanskaya
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
            </p>
          </div>
          <div
            class="MuiBox-root css-1r3t2am"
          >
            <img
              alt="card"
              src="konstantin_shumak.jpg"
              style="max-width: 400px; flex: 1; margin-bottom: 24px; object-fit: cover; border-radius: 16px;"
            />
            <h4
              class="MuiTypography-root MuiTypography-h4 css-1psfiy1-MuiTypography-root"
            >
              Konstantin Shumak
            </h4>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1no9wdb-MuiTypography-root"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis laoreet tellus.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1e5i6q7"
    >
      <div
        class="MuiBox-root css-1mioigq"
      >
        <svg
          data-file-name="SvgGhIcon"
        />
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-j3poeh-MuiTypography-root-MuiLink-root"
          href="https://github.com/in-my-book-team"
          target="_blank"
        >
          GitHub
        </a>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1jx4n4e-MuiTypography-root"
      >
        Copyright © 2023. All rights reserved
      </p>
    </div>
  </div>
</div>
`;
